e2e-tests/test/hardhat-apis.test.ts

```diff
@@ -84,3 +88,64 @@ xdescribe("hardhat_impersonateAccount & hardhat_stopImpersonatingAccount", funct
     expect(await provider.getBalance(RichAccounts[0].Account)).to.equal(beforeBalance.sub(0.42));
   });
 });
+
+describe("hardhat_setCode", function () {
+  it.only("Should set code at an address", async function () {
+    // Arrange
+    const wallet = new Wallet(RichAccounts[0].PrivateKey);
+    const deployer = new Deployer(hre, wallet);
+
+    const address = "0x1000000000000000000000000000000000001111";
+    const artifact = await deployer.loadArtifact("Return5");
+    const contractCode = [...ethers.utils.arrayify(artifact.deployedBytecode)];
+
+    // Act
+    await provider.send("hardhat_setCode", [address, contractCode]);
+
+    // Assert
+    const result = await provider.send("eth_call", [
+      {
+        to: address,
+        data: keccak256(ethers.utils.toUtf8Bytes("value()")).substring(0, 10),
+        from: wallet.address,
+        gas: "0x1000",
+        gasPrice: "0x0ee6b280",
+        value: "0x0",
+        nonce: "0x1",
+      },
+      "latest",
+    ]);
+    expect(BigNumber.from(result).toNumber()).to.eq(5);
+  });
+});
+
+describe("hardhat_setCode", function () {
```

Comment:
No need for a separate `describe`, you can have multiple `it(...)` inside of a `describe`

Reply:
I generally avoid having multiple `it` inside a `describe` as most `describe`s run in isolation but I guess that's not the case here so we can have them together.